### PR TITLE
[VersionControl] Add validations adding a new remote (Git)

### PR DESCRIPTION
**Pull request by jsuarezruiz**
Add URL validation adding a new remote in Git.

Fixes gh #7114

### CODE REVIEW

File: main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/EditRemoteDialog.cs [hunk 1]
```diff
@@ -107,5 +108,10 @@ protected void OnEntryPushUrlChanged (object sender, System.EventArgs e)
 
 			UpdateButtons ();
 		}
+
+		bool IsValidUrl (string url)
+		{
+			return Uri.TryCreate (url, UriKind.Absolute, out var dummy);
```
Comment (Therzok) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/EditRemoteDialog.cs]:
```suggestion
			return Uri.TryCreate (url, UriKind.Absolute, out _);
```

Comment (netonjm) [reply main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/EditRemoteDialog.cs]: I would include `entryUrl.Text.Length > 0` check in  `bool IsValidUrl (string url)`

Comment (Therzok) [reply main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/EditRemoteDialog.cs]: TryCreate handles empty string, afaik.

Comment (netonjm) [reply main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/EditRemoteDialog.cs]: then kill!

Comment (jsuarezruiz) [reply main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/EditRemoteDialog.cs]: Yes, `TryCreate` will return false if the url string is empty.

File: main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/EditRemoteDialog.cs [hunk 2]
```diff
@@ -80,7 +81,7 @@ void SetPushUrlTextStyle (bool disabled)
 
 		void UpdateButtons ()
 		{
-			buttonOk.Sensitive = entryName.Text.Length > 0 && entryUrl.Text.Length > 0 && entryPushUrl.Text.Length > 0;
+			buttonOk.Sensitive = entryName.Text.Length > 0 && IsValidUrl (entryUrl.Text) && entryPushUrl.Text.Length > 0;
```
Comment (Therzok) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/EditRemoteDialog.cs]:
```suggestion
			buttonOk.Sensitive = entryName.Text.Length > 0 && IsValidUrl (entryUrl.Text) && IsValidUrl (entryPushUrl) > 0;
```

Comment (sevoku) [reply main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/EditRemoteDialog.cs]:
```suggestion
			buttonOk.Sensitive = entryName.Text.Length > 0 && IsValidUrl (entryUrl.Text) && IsValidUrl (entryPushUrl);
```

**Comment by jsuarezruiz** [comment]
I have reviewed and escaped an important detail, in Git different procols are supported. In addition, `GitRepositry` class already had a `IsUrlValid` method that validated the content in addition to the supported protocols. Now we use that method to validate the content of `entryUrl` and `entryPushUrl`.

File: main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/EditRemoteDialog.cs (hunk 2 above)
Comment (jsuarezruiz) [reply main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/EditRemoteDialog.cs]: Please, review the latest change and latest comment.

**Pull request merged**